drivers/nrf24l01p/nrf24l01p_communication.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup drivers_nrf24l01p
+ * @{
+ *
+ * @file
+ * @brief     Functions to communicate with the NRF24L01P transceiver via SPI
+ *
+ * @author Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+#include <errno.h>
+#include <string.h>
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "xtimer.h"
+#include "nrf24l01p_constants.h"
+#include "nrf24l01p_communication.h"
+
+#define ENABLE_DEBUG    (0)
+
+#ifndef MIN
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+#endif
+
+#define SPI_BUS     (dev->params.spi)
+#define SPI_PIN_CS  (dev->params.pin_cs)
+#define DUMMY       NULL
+
+#define NRF24L01P_REG_MAX_WIDTH NRF24L01P_MAX_ADDR_WIDTH
+
+static void swap_bytes_inplace(void *buffer, size_t len)
+{
+    uint8_t *bytes = (uint8_t *)buffer;
+
+    for (size_t i = 0; i < len / 2; i++) {
+        uint8_t tmp = bytes[i];
+        bytes[i] = bytes[len - 1 - i];
+        bytes[len - 1 - i] = tmp;
+    }
+}
+
+int nrf24l01p_acquire(nrf24l01p_t *dev)
+{
+    DEBUG("[nrf24l01p] acquire ...\n");
+    int a = spi_acquire(dev->params.spi, dev->params.pin_cs, SPI_MODE_0,
+                        dev->params.spi_clk);
+    #ifndef NDEBUG
+    dev->have_spi_access = 1;
+    #endif
```

Comment:
```C
if (IS_USED(ENABLE_DEBUG)) {
    dev->have_spi_access = 1;
}
```

`NDEBUG` is usually set globally. `ENABLE_DEBUG` is used on a per-file level, which allows to reducing the overhead when not used.

Reply:
If `IS_USED(ENABLE_DEBUG)` is true, will I be able to use `assert()`?
Because I just assert that I have SPI access in `netdev_t::nrf24l01p_recv`.
I did not want to call `nrf24l01p_acquire()` there because `nrf24l01p_recv` may be called from `nrf24l01p_isr` and I don´t want to risk that I have to block in that case.

Reply:
`assert()` is always available. If `NDEBUG` is defined, it is however a NOP.

But I guess your question is whether it could happen that the `assert()` is triggered because the `dev->have_spi_access = 1` was deactivated. That would of course something to be avoided.

But I don't think that this check will be needed anymore once the driver is ready for merging. It is however often a good idea to keep debug code around and only disable it, so that future debugging sessions can just use that again.

Reply:
Right now I am substituting `NDEBUG` with `IS_USED(ENABLE_DEBUG)` and it turns out that it is a bit tedious because `ENABLE_DEBUG` is not global and `assert()` and the declaration of the variable that is asserted are not in the same file necessarily.

I could keep the `NDEBUG` until it gets merged.
Or, I could use some header `nrf24l01p_debug.h` that forces `ENABLE_DEBUG` to be `1` if
`NDEBUG` is not defined.
Or I could remove the code with `NDEBUG` right now.